Tella/Data/Networking/Repositories/GDriveRepository.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  GDriveRepository.swift
+//  Tella
+//
+//  Created by gus valbuena on 5/28/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import GoogleSignIn
+import GoogleAPIClientForREST
+import Combine
+
+protocol GDriveRepositoryProtocol {
+    func handleSignIn() async throws -> Void
+    func restorePreviousSignIn() async throws -> Void
+    func handleUrl(url: URL)
+    func getSharedDrives() -> AnyPublisher<[SharedDrive], Error>
+}
+
+class GDriveRepository: GDriveRepositoryProtocol  {
+    static let shared = GDriveRepository()
+    
```

Comment:
What do you think about removing the shared and use DI instead?

Reply:
Yeah that's a good idea. My first approach was to use dependency injection but when I tried to implement it before on resources we talked and we decide to not use it for sake of the code consistency. @dhekra-rouatbi what do you think on implement it for this repository?

Reply:
I think maintaining code consistency will lead to a complex codebase if we follow the previous implementation?

Reply:
I agree on implementing DI for this but let's discuss it on tomorrow's standup so we take a common decision